test/functional/shared.js

```diff
@@ -57,8 +98,26 @@ function setupDatabase(configuration, dbsToClean) {
     );
 }
 
-function makeCleanupFn(client) {
-  return function(err) {
+/**
+ * Safely perform a test with provided MongoClient, ensuring client won't leak.
+ *
+ * @param {MongoClient} [client] if not provided, withClient must be bound to test function `this`
```

Comment:
while we're here, could the first parameter also be a connection string passed to `this.configuration.newClient(connStr)`? I see cases below where you were doing `withClient(this.configuration.newClient('string'), ...`